iOSClient/AppDelegate.swift

```diff
@@ -285,6 +293,29 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
         NCCommunicationCommon.shared.writeLog("bye bye")
     }
     
+    // MARK: - Privacy Protection
+       
+    private func showPrivacyProtectionWindow() {
+        
+        if !CCUtility.getPrivacyScreen() { return }
+        if CCUtility.isPasscodeEnableAtStart() { return }
```

Comment:
isPasscodeEnableAtStart verify passcode + Do not ask at startup

Reply:
**Setup**:
- Lock: On
- Do not ask...: Off
- Enable splash: On

**Expected**: Show splash screen when inactive

**Actual**: `CCUtility.isPasscodeEnableAtStart()` causes return. No splash is displayed!

Reply:
exact ! No splash because is displayed the passcode in background

Reply:
passcode WINS

Reply:
Well, *imo* it should still display the splash screen in that case, regardless of if a passcode is active. And when becoming active dismiss splash show the passcode.
But if not, it should at least display the passcode right away on `applicationWillResignActive`, shouldn't it? Because if I have splash active I expect everything to be hidden 'instantly' not only *later* when in background.

Reply:
This is true, but , sincerely, it's horrible... the splash screen arrive when the app is not active ... after remove when enter in background (wins the passcode) ... don't like the effect (try it with iPhone) ... but we can forced the switch in Settings ... if passcode -> automatically splash in off ....

Reply:
I'm not sure what you mean with horrible. If you mean the splash screen that's shown right before the passcode for a split second, you can fix that easily by just calling `hidePrivacyProtectionWindow()` from `presentPasscode()`.

Only passcode (as it is now):
https://user-images.githubusercontent.com/18512366/144263965-a36d8aba-b9ed-427f-8f26-0fc9c8d93609.MP4

Splash screen & passcode (the suggestion):
https://user-images.githubusercontent.com/18512366/144263995-89a3a912-69ad-456c-bd2c-4824410db228.MP4

Reply:
ok, try